share-services/src/main/resources/alfresco/templates/webscripts/org/alfresco/slingshot/node-browser/node-browser.get.json.ftl

```diff
@@ -1,32 +1,76 @@
 <#macro dateFormat date>${date?string("dd MMM yyyy HH:mm:ss 'GMT'Z '('zzz')'")}</#macro>
 <#escape x as jsonUtils.encodeJSONString(x)>
 <#macro printPropertyValue p>
-   <#if p.value??>
-      <#if p.value?is_date>
+   <#attempt>
+      <#if p.value??>
+        <#if p.value?is_date>
          "<@dateFormat p.value />"
-      <#elseif p.value?is_boolean>
+        <#elseif p.value?is_boolean>
          ${p.value?string}
-      <#elseif p.value?is_number>
+        <#elseif p.value?is_number>
          ${p.value?c}
-      <#elseif p.value?is_string>
+        <#elseif p.value?is_string>
          "${p.value}"
-      <#elseif p.value?is_hash>
-         <#assign result = "{"/>
-         <#assign first = true />
-         <#list p.value?keys as key>
-            <#if first = false>
-               <#assign result = result + ", "/>
-            </#if>
-            <#assign result = result + "${key} = ${p.value[key]}" />
-            <#assign first = false/>
-         </#list>
-         <#assign result = result + "}"/>
-         <#-- output the result -->
-         "${result}"
+        <#elseif p.value?is_hash || p.value?is_enumerable>
+            <#assign val>
+               <@convertToJSON p.value />
+            </#assign>
+            "${val}"
+        </#if>
+   	  <#else>
+   	     null
       </#if>
-   <#else>
-      null
-   </#if>
+   <#recover>
+      "${.error}"
+   </#attempt>
+</#macro>
+<#macro convertToJSON v>
+   <#attempt>
+      <#if v??>
+         <#if v?is_date>
+            "<@dateFormat v />"
+         <#elseif v?is_boolean>
+            ${v?string}
+         <#elseif v?is_number>
+            ${v?c}
+         <#elseif v?is_string>
+            "${v?string}"
+         <#elseif v?is_hash>
+            <@compress single_line=true>
+               {
+               <#assign first = true />
+               <#list v?keys as key>
+               <#if first = false>,</#if>
+               "${key}":
+               <#if v[key]??>
+                  <@convertToJSON v[key] />

```

Comment:
Worried about possible performance issues in case of large serialised objects, or perhaps even DoS attacks in case of a specially crafted object for this purpose (?).